retrieval/discovery/ec2.go

```diff
@@ -110,12 +115,25 @@ func (ed *EC2Discovery) refresh() (*config.TargetGroup, error) {
 				labels := model.LabelSet{
 					ec2LabelInstanceID: model.LabelValue(*inst.InstanceId),
 				}
+				labels[ec2LabelPrivateIP] = model.LabelValue(*inst.PrivateIpAddress)
+				addr := fmt.Sprintf("%s:%d", *inst.PrivateIpAddress, ed.port)
+
 				if inst.PublicIpAddress != nil {
 					labels[ec2LabelPublicIP] = model.LabelValue(*inst.PublicIpAddress)
+					labels[ec2LabelPublicDns] = model.LabelValue(*inst.PublicDnsName)
+					// replace address with public dns name if present
+					addr = fmt.Sprintf("%s:%d", *inst.PublicDnsName, ed.port)
 				}
-				labels[ec2LabelPrivateIP] = model.LabelValue(*inst.PrivateIpAddress)
-				addr := fmt.Sprintf("%s:%d", *inst.PrivateIpAddress, ed.port)
 				labels[model.AddressLabel] = model.LabelValue(addr)
+
+				labels[ec2LabelAz] = model.LabelValue(*inst.Placement.AvailabilityZone)
+
+				if inst.VpcId != nil {
+					labels[ec2LabelVpcId] = model.LabelValue(*inst.VpcId)
+					// take the subnet id of eth0, discard all secondary interfaces
+					labels[ec2LabelSubnetId] = model.LabelValue(*inst.NetworkInterfaces[0].SubnetId)
```

Comment:
Someone may have multiple ENIs, personally I am not sure how to present that information in a usable way. I am assuming that when someone will need that, he/she will have a clear idea of what he needs and how to implement this.

Do you have any idea/guidance on the label structure for this case?

Thanks